Add privacy wiki, coinjoins.org, various research papers and coinjoin history overview (#1614)
bitcoinops/bitcoinops.github.io: _topics/en/coinjoin.md

  - title: "Zerolink: the Bitcoin fungibility framework"
    link: https://github.com/nopara73/ZeroLink

>   - title: "Privacy Privacy Wiki"

Reviewer: I think these are secondary sources of aggregated information as opposed to primary, original sources.

    link: https://en.bitcoin.it/wiki/Privacy

>   - title: "Coinjoins.org - Learn about collaborative bitcoin transactions"

Reviewer: Maybe "see_also"?

see_also:
  - title: Payjoin
    link: topic payjoin

>     

Reviewer: ```suggestion

```

    link: topic payjoin
  - title: "Coinjoin (Bitcoin Wiki: Privacy)"
    link: https://en.bitcoin.it/Privacy#CoinJoin

>   - title: "Privacy Privacy Wiki"

Reviewer 2: Duplicate word.

    link: https://en.bitcoin.it/wiki/Privacy
  - title: "Coinjoins.org - Learn about collaborative bitcoin transactions"
    link: https://coinjoins.org/

---

> On July 2nd 2011, the Bitcointalk user [Hashcoin][hashcoin ref] first described the concept of a collaborative transaction to achieve privacy. Later in 2013, the concept was named a coinjoin after a [proposal][gmaxwell coinjoin] by Gregory Maxwell. He laid out the ideas and set up a developer bounty to make them a reality.

Reviewer 2: ```suggestion
On July 2nd 2011, the Bitcointalk user [Hashcoin][hashcoin ref] first described the concept of a collaborative transaction to achieve privacy. Later in 2013, the concept was named coinjoin after a [proposal][gmaxwell coinjoin] by Gregory Maxwell. He laid out the ideas and set up a developer bounty to make them a reality.
```

    link: topic payjoin
  - title: "Coinjoin (Bitcoin Wiki: Privacy)"
    link: https://en.bitcoin.it/Privacy#CoinJoin

>   - title: "TumbleBit: An Untrusted Bitcoin-Compatible Anonymous Payment Hub"

Reviewer: Seems like this might be a primary source?

Reviewer 3: It is a standalone research paper, so imo yes.

    link: https://eprint.iacr.org/2016/575.pdf

>   - title: "ZeroLink: The Bitcoin Fungibility Framework"

Reviewer: We already have this link above and can drop it from here

    link: https://github.com/nopara73/ZeroLink

>   - title: "WabiSabi: Centrally Coordinated Coinjoins with Variable Amounts"

Reviewer: If this is the primary source information for the wabisabi coinjoin protocol, I think it could be under primary sources.

Reviewer 3: It is the primary.

    link: https://github.com/zkSNACKs/WabiSabi
  
---

> On July 2nd 2011, the Bitcointalk user Hashcoin first described the concept of a collaborative transaction to achieve privacy. Later in 2013, the concept was named a coinjoin after a [proposal][gmaxwell coinjoin] by Gregory Maxwell. He laid out the ideas and set up a developer bounty to make them a reality.

Reviewer: Is there a good link for the hashcoin post?

Author: Added

In November 2016, an alternative to coinjoins was release with Tumblebit. Users could create two fixed-amount payment channels to a Tumbler (coordinator) who can't steal their coins or deanonymize them. The coins get sent back to a user from payment channels of other users. It takes a total of 4 transactions to complete, but you get a very high anonymity rate. The wallets Breeze and Hidden Wallet (now Wasabi Wallet) were the main clients to use this privacy technique.

On August 14th 2017, the ZeroLink paper was introduced as The Bitcoin Fungibility Framework. It was a collaboration between the Hidden Wallet (now Wasabi Wallet) and Samourai Wallet teams. The research paper was authored by Adam Ficsor (Nopara73) with some support from TDev. It provided a framework not only for a coinjoin protocol, but for privacy wallets in general. 

On October 31st 2018, Wasabi Wallet 1.0 was officially released publicly. It provided an efficient alternative to Joinmarket for less technical users and a full zero-link coinjoin implementation. Like Tumblebit, there was no need to trust the central coordinator. Unlike Tumblebit, you could get anonymity very quickly and for a low fee.

> On June 25th 2019, the Samourai Wallet team launched a coinjoin feature called Whirlpool, which implements the Zerolink protocol with one major change. Instead of registering non-private UTXOs as inputs in a coinjoin transaction and getting the exceeding change as a non-private output, a premixing transaction takes place beforehand. In it, excess change is separated from UTXOs to mix and from the coordinator fee. Subsequently, UTXOs to mix would form a coinjoin transaction that produces no toxic change outputs.

Reviewer: Perhaps a definition of toxic change could be helpful

> In February 2021, the WabiSabi paper was released after many months of open discussions on the development of a third-generation coinjoin protocol that would solve the predecessor issues.By introducing cryptographic methods, the fixed coinjoin output limit was removed and it was now possible to decompose the total bitcoin into variable output amounts. This massively reduced the occurrence of toxic change and made WabiSabi the most efficient coinjoin protocol ever invented.

Reviewer 3: ```suggestion
In February 2021, the WabiSabi paper was released after many months of open discussions on the development of a third-generation coinjoin protocol that would solve the predecessor issues. By introducing anonymous amount eCash tokens, the fixed coinjoin output limit was removed and it was now possible to decompose the total bitcoin into variable output amounts. This massively reduced the occurrence of toxic change and made WabiSabi the most efficient coinjoin protocol ever invented.
```

Missing space, and minor clarification.
